### PR TITLE
New Caching Client, Improvements in gateway

### PR DESCRIPTION
**Pull request by GerardSmit**
Removed all the caching from Miki.Discord and moved this into Miki.Discord.Caching. This way we don't have the writing and reading cache items in seperated projects.

### CODE REVIEW

**Comment by GerardSmit** [comment]
I still want to do some testing before this gets merged.

Tomorrow I'll set-up a test client to test if everything works as intended.

**Comment by GerardSmit** [comment]
I'll fix the merge conflicts and change `IReadOnlyList` to `IEnumerable` as discussed in #19.

File: Miki.Discord.Common/Models/IApiClient.cs [hunk 1]
```diff
@@ -45,23 +45,23 @@ public interface IApiClient : IGatewayApiClient
 
 		Task<DiscordChannelPacket> GetChannelAsync(ulong channelId);
 
-		Task<List<DiscordChannelPacket>> GetChannelsAsync(ulong guildId);
+		Task<IReadOnlyList<DiscordChannelPacket>> GetChannelsAsync(ulong guildId);
```
Comment (velddev) [inline Miki.Discord.Common/Models/IApiClient.cs]: Regarding these; are we going for eumerables to then move to IAsyncEnumerables? or keep them as lists for now?

Comment (GerardSmit) [reply Miki.Discord.Common/Models/IApiClient.cs]: I've changed them to enumerables.

**Comment by GerardSmit** [comment]
Done 🎉

If you're okay with the latest changes then you have the honor to click the "Merge pull request" button.

**Pull request merged**